File: Workshop-main/Cross-Platform-Workshop-Hub-Mobile/lib/model/userEditRequest.dart
// ignore_for_file: file_names
import 'package:json_annotation/json_annotation.dart';
part 'userEditRequest.g.dart';
// ignore_for_file: non_constant_identifier_names
@JsonSerializable()
class UserEditRequest {
  String full_name;
  String user_name;
  String email;
  String phoneNumber;
  double balance;
  String image_url;
  bool isEnable;
  List<UserAddress> userAddresses;
  List<UserBank> userBanks;

  UserEditRequest({
    required this.full_name,
    required this.user_name,
    required this.email,
    required this.phoneNumber,
    required this.balance,
    required this.image_url,
    required this.userAddresses,
    required this.userBanks,
    required this.isEnable,
  });

   factory UserEditRequest.fromJson(Map<String, dynamic> json) => _$UserEditRequestFromJson(json);
  Map<String, dynamic> toJson() => _$UserEditRequestToJson(this);
}
@JsonSerializable()
class UserAddress {
  int id;
  String Address;
  String City;
  String State;
  int PostalCode;

  UserAddress({
    required this.id,
    required this.Address,
    required this.City,
    required this.State,
    required this.PostalCode,
  });

  factory UserAddress.fromJson(Map<String, dynamic> json) =>
      _$UserAddressFromJson(json);
  Map<String, dynamic> toJson() => _$UserAddressToJson(this);
}
@JsonSerializable()
class UserBank {
  int id;
  String bankName;
  String bankAccount;

  UserBank({
    required this.id,
    required this.bankName,
    required this.bankAccount,
  });

  factory UserBank.fromJson(Map<String, dynamic> json) =>
      _$UserBankFromJson(json);
  Map<String, dynamic> toJson() => _$UserBankToJson(this);

}

File: Workshop-main/Cross-Platform-Workshop-BackEnd/src/main/java/com/workshop/dto/useDTO/UserEditRequest.java
package com.workshop.dto.useDTO;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.annotation.Nullable;
import lombok.*;
import lombok.experimental.Accessors;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Accessors(chain = true)
public class UserEditRequest {
    private String full_name;
    private String user_name;
    private String email;
    private String phoneNumber;
    private Double balance;
    private String image_url;
    @JsonProperty("userAddresses")
    @Nullable
    @Builder.Default
    private List<UserAddress> userAddresses = Collections.emptyList();
    @JsonProperty("userBank")
    @Nullable
    @Builder.Default
    private List<UserBank> userBanks = Collections.emptyList();
    @Getter
    @Setter
    @AllArgsConstructor
    @NoArgsConstructor
    public static class UserAddress {
        private Long id;
        private String Address;
        private String City;
        private String State;
        private int PostalCode;
    }
    @Getter
    @Setter
    @AllArgsConstructor
    @NoArgsConstructor
    public static class UserBank {
        private Long id;
        private String bankName;
        private String bankAccount;
    }

}
